src/main/java/hexgui/gui/MessageDialogs.java

```diff
@@ -0,0 +1,390 @@
+// MessageDialogs.java
+
+package hexgui.gui;
+
+import static hexgui.gui.GuiUtil.insertLineBreaks;
+
+import hexgui.util.Platform;
+import hexgui.util.PrefUtil;
+import hexgui.util.StringUtils;
+import java.awt.Component;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.prefs.Preferences;
+import javax.swing.Box;
+import javax.swing.JCheckBox;
+import javax.swing.JComponent;
+import javax.swing.JDialog;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+
+/** Simple message dialogs. */
+public final class MessageDialogs {
+  public MessageDialogs(String applicationName) {
+    m_applicationName = applicationName;
+  }
+
+  public void showError(Component frame, String mainMessage, String optionalMessage) {
+    showError(frame, mainMessage, optionalMessage, true);
+  }
+
+  public void showError(
+      Component frame, String mainMessage, String optionalMessage, boolean isCritical) {
+    int type;
+    if (isCritical) type = JOptionPane.ERROR_MESSAGE;
+    else type = JOptionPane.PLAIN_MESSAGE;
+    Object[] options = {"Close"};
+    Object defaultOption = options[0];
+    String title = "Error" + " - " + m_applicationName;
+    show(
+        null,
+        frame,
+        title,
+        mainMessage,
+        optionalMessage,
+        type,
+        JOptionPane.DEFAULT_OPTION,
+        options,
+        defaultOption,
+        -1);
+  }
+
+  public void showError(Component frame, String message, Exception e) {
+    showError(frame, message, e, true);
+  }
+
+  public void showError(Component frame, String message, Exception e, boolean isCritical) {
+    showError(frame, message, StringUtils.getErrorMessage(e), isCritical);
+  }
+
+  public void showInfo(
+      Component frame, String mainMessage, String optionalMessage, boolean isCritical) {
+    showInfo(null, frame, mainMessage, optionalMessage, isCritical);
+  }
+
+  public void showInfo(
+      String disableKey,
+      Component frame,
+      String mainMessage,
+      String optionalMessage,
+      boolean isCritical) {
+    if (checkDisabled(disableKey)) return;
+    int type;
+    if (isCritical) type = JOptionPane.INFORMATION_MESSAGE;
+    else type = JOptionPane.PLAIN_MESSAGE;
+    Object[] options = {"Close"};
+    Object defaultOption = options[0];
+    String title = "Information" + " - " + m_applicationName;
+    show(
+        disableKey,
+        frame,
+        title,
+        mainMessage,
+        optionalMessage,
+        type,
+        JOptionPane.DEFAULT_OPTION,
+        options,
+        defaultOption,
+        -1);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 71:88, 160:177.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.